CMakeLists.txt

```diff
@@ -1,29 +1,38 @@
-cmake_minimum_required(VERSION 2.8.12)
+#TODO
+# - fix / modernize cmake
+# - lower cmake version a bit ~3.7 to support older systems
+# - fix abseil warnings or try to use SYSTEM / -isystem includes
+
+cmake_minimum_required(VERSION 3.8)
```

Comment:
Separate PR for increasing cmake min version.  

What's the advantage of doing this? Removing version >= 3.8.0 checks?

Reply:
2.8.x is really too old it does not really support targets (even though they became available in 2.8.11 iirc) and it does not have support for newer standards and language features. I think it is best to lower the version to something that still supports at least c++14, but something else might be ok. It could be better to use something that is available in the current Ubuntu release or in Travis containers. Eventually it is up to you! Please don't insist to support CentOS  6:)

Reply:
It looks like 3.8.0 is only a year old. That seems a bit new to require.

Maybe something in the 3.4-3.7 range? What's the oldest version that supports the features you want to use?